feat: разработано аудио приложение

seregga opened the pull request:
Аудио приложение со звуками природы, можно
включать, выключать и ставить на паузу звуки.
А так же есть регулировка громкости.

Задание #1 из блока вебпак

seregga committed:
feat: разработано аудио приложение

Аудио приложение со звуками природы, можно
включать, выключать и ставить на паузу звуки.
А так же есть регулировка громкости.

Задание #1 из блока вебпак

antisedative commented on webpack.config.prod.js:
```diff
@@ -0,0 +1,13 @@
+
+module.exports = merge(commonConfig, {
+    mode: 'production',
+    devtool: 'source-map',
```
devtool лучше убрать из прод конфига

seregga replied:
Сделано

antisedative commented on webpack.config.common.js:
```diff
@@ -0,0 +1,61 @@
+        test: /\.(png|jpg|svg)$/,
+        type: 'asset/resource',
+      },
+    //   {
+    //     test: /\.(mp3)$/,
+    //     type: 'asset/resource',
+    //   },
```
помним про чистоту кода в пул реквестах, комментариев и логов быть не должно)

seregga replied:
Сделано

antisedative commented on src/index.js:
```diff
@@ -0,0 +1,93 @@
+function createMusicList(list) {
+    const musicList = document.getElementsByTagName('ul')[0];
+
+    list.forEach(el => {
+        const track = document.createElement('li')
+        track.style.background = getBackStr(el.background)
+
+        const trackBtn = document.createElement('span')
+        trackBtn.style.background = getBackStr(el.icon)
+        trackBtn.setAttribute('id', `${el.id}`)
+
+        track.append(trackBtn)
+        musicList.append(track)
+
+        trackBtn.addEventListener('click', (e) => handleTrackBtn(e, el))
+    })
```
слушатель событий создается в цикле, было бы тут пару тысяч элементов - были бы проблемы)
в этом задании можно обойтись всего одним слушателем сразу для всех блоков погоды

seregga replied:
Сделано

seregga committed:
perf: правки по производительности плеера

слушатель событий вынесен из цикла и присвоен
блоку ul

Задание #1 из блока вебпак

seregga committed:
chore:  убран devtool и удалены комментарии

Задание #1 из блока вебпак
